tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_different_function_signatures.bal

```diff
@@ -326,6 +326,24 @@ function testMethodVarargEvaluationCount() {
     assertValueEquality(1, counter);
 }
 
+type Func function ();
+
+public function f1() {
+    json|error res = foo:testFunc("f1");
+}
+
+final readonly & map<Func> func = {
+    "f1": f1
```

Comment:
I think this should ideally work even if `f1` calls `testFunc` defined in this module - https://github.com/ballerina-platform/ballerina-lang/issues/30042.

Shall we add another test where it is calling the function?